### PR TITLE
scheduler - use an LRU cache to track taskDurations

### PR DESCRIPTION
**Pull request by dgdonut**
Address root cause by converting taskDurations to an LRU that will cap at a preset size, instead of a map that grows infinitely or has to be manually pruned.

### CODE REVIEW

**Comment by dgdonut** [comment]
edited: test updated

File: scheduler/server/job_state.go [hunk 1]
```diff
@@ -4,7 +4,9 @@ import (
 	"math"
 	"time"
 
+	lru "github.com/hashicorp/golang-lru"
 	log "github.com/sirupsen/logrus"
+
```
Comment (anthonyjpratti) [inline scheduler/server/job_state.go:9]: nit: blank line

File: scheduler/server/stateful_scheduler.go [hunk 2]
```diff
@@ -123,9 +129,10 @@ type averageDuration struct {
 	duration time.Duration
 }
 
-func (ad *averageDuration) update(d time.Duration) {
+func (ad *averageDuration) update(d time.Duration) *averageDuration {
 	ad.count++
 	ad.duration = ad.duration + time.Duration(int64(d-ad.duration)/ad.count)
+	return ad
```
Comment (anthonyjpratti) [inline scheduler/server/stateful_scheduler.go]: I don't see this used anywhere,  maybe remove?

**Review by anthonyjpratti — APPROVED** [review]

File: scheduler/server/job_state.go (hunk 1 above)
Comment (dgdonut) [reply scheduler/server/job_state.go:9]: added for consistency with other files (separate twitter/scoot pkgs from 3rdparty pkgs)

File: scheduler/server/stateful_scheduler.go (hunk 2 above)
Comment (dgdonut) [reply scheduler/server/stateful_scheduler.go]: thanks, added for previous iteration, not needed anymore

**Pull request merged**